src/Service/ApplicationFactory.php

```diff
@@ -27,11 +27,22 @@ class ApplicationFactory implements FactoryInterface
      */
     public function __invoke(ContainerInterface $container, $name, array $options = null)
     {
-        return new Application(
+        $application = new Application(
             $container,
             $container->get('EventManager'),
             $container->get('Request'),
             $container->get('Response')
         );
+
+        if (! $container->has('config')) {
+            return $application;
+        }
+
+        $em = $application->getEventManager();
+        $listeners = $container->get('config')[Application::class]['listeners'] ?? [];
+        foreach ($listeners as $listener) {
+            $container->get($listener)->attach($em);
+        }
```

Comment:
This answers part of my question about the changes to `Application::bootstrap()`. However, you need to document this in the changelog and the manual.

Reply:
This is meant more of a placeholder at this point that is meant to preserve the functionality. Event system would need to have a more detailed revision and potential utilization of PSR-14 would change this part entirely.

One of the reasons why I opted for skipping documenting the changes in CHANGELOG and providing updated documentation by this PR.